Lib/asyncio/runners.py

```diff
@@ -25,7 +32,11 @@ class Runner:
     and properly finalizes the loop at the context manager exit.
 
     If debug is True, the event loop will be run in debug mode.
-    If loop_factory is passed, it is used for new event loop creation.
+    If loop_factory is passed, it is used for new event loop creation, otherwise
+    a ProactorEventLoop will be started on Windows or a SelectorEventLoop will
+    be started on unix
+    If set_policy_loop is True, the event loop in the default policy will be
+    set.
```

Comment:
I think it's important to maintain the asyncio.Runner support for a high level policy-free use of asyncio, so that the policy system can be deprecated and removed

Reply:
By adding this option we make it *more difficult* to deprecate the policy system because then this option will need to be deprecated as well. Users who specify this option will have more to do to migrate.

Reply:
> With this flag disabled, we retain the current behaviour which causes issues. I would guess that most people don't fully understand what it does and will leave it to the default, then potentially run into these issues.

It should be opt in for asyncio.Runner

Reply:
yep: set_policy_loop defaults to False in `asyncio.Runner`: `def __init__(self, *, debug=None, loop_factory=None, set_policy_loop=False):`

people using the old `asyncio.run` get the old behaviour that uses the policy loop factory and sets the policy loop. People using the new `asyncio.Runner` get the new behaviour which does not use the policy loop factory or set the policy loop by default